### ISSUE TITLE
ISO19115-3 mappings to OARec

### ISSUE THREAD

**Issue by ByronCinNZ** [opengeospatial/ogcapi-code-sprint-2021-07#16]
Candidate crosswalk attached. Needs discussion.
[iso19115OARec-Crosswalk.csv](https://github.com/opengeospatial/ogcapi-code-sprint-2021-07/files/6862516/iso19115OARec-Crosswalk.csv)

**Comment by ghobona** [comment]
@ByronCinNZ Thanks for this! Very much appreciated.

**Comment by ByronCinNZ** [comment]
I did not complete the "links" mappings due to my lack of understanding. However I have a few mappings I think work

**Comment by ByronCinNZ** [comment]
What else would belong in "links:"? mdb:metadataStandard? mdb:metadataProfile?

**Comment by ByronCinNZ** [comment]
Candidate new mapping from ISO19115-3 Spatial Resolution `//mdb:MD_Metadata  /mdb:identificationInfo   /mri:MD_DataIdentification /mri:spatialResolution /mri:MD_Resolution`
(May be represented in a number of ways? Restricting it to numerical representation would aid search) 
Candidate geoJSON mapping - `properties.resolution`

**Comment by pvretano** [comment]
@ByronCinNZ I can't seem to read the .xlsx file.  I'm using libreoffice.  Can you save it as an .xls file instead?

**Comment by ByronCinNZ** [comment]
@pvretano Try this
[iso19115OARec-Crosswalk.xls](https://github.com/opengeospatial/ogcapi-code-sprint-2021-07/files/6865005/iso19115OARec-Crosswalk.xls)

**Comment by pvgenuchten** [comment]
Great work Byron!

Some comments/questions:

- for type you refer to topiccategory, isn't this element more fit to capture hierarchylevel (e.g. dataset, service, sensor, document)
- links section needs further clarification, ogcapi-common provides a number of types of links, we should capture (only some are related to this crosswalk):
   - links to related records
   - a link to the original location of the record (point of truth, canonical url)
   - a link to the collection where this record is a part of
   - link to alternative representations of this record (profile, language, format)
- shouldn't format be a part of the ./associations/@type="application/geopackage+sqlite" (i missed that this is now a separate element)

**Comment by pvgenuchten** [comment]
Notice that https://github.com/geopython/pygeoapi/blob/master/tests/load_tinydb_records.py contains a mapping of iso19139:2007 to ogc-api-records

**Comment by ByronCinNZ** [comment]
@pvgenuchten Thanks for the feedback Paul.
I will change `type` to `metadataScope` and replace my current mapping of `theme` with `type`
For `links` I have included references for a link to the original location of the record (point of truth, canonical url) and link to alternative representations of this record (profile, language, format)
I also include `parentMetadata` mapping in `links`.

**Comment by ByronCinNZ** [comment]
Updated mappings are attached here 
[iso19115OARec-Crosswalk.xls](https://github.com/opengeospatial/ogcapi-code-sprint-2021-07/files/6866139/iso19115OARec-Crosswalk.xls)

**Comment by pvgenuchten** [comment]
Suggestion could be to extend this a bit, add columns for iso19139 and DCAT, e.g.

| OGC-API Records | iso19139:2007 | iso19115-3:2012 | DCAT 2 |
| --- | --- | --- | --- |
| recordId  | //gmd:fileIdentifier | //mcc:MD_Identifier | //dct:identifier |
| recordCreated | - | //mdb:dateInfo[@dateType='creation'] |  |
| recordUpdated | gmd:dateStamp | //mdb:dateInfo[@dateType='LastUpdate'] | dct:modified |
| type | //gmd:hierarchyLevel |  | dct:type| |
| title | //gmd:title | //cit:title | //dct:title |
| description | //gmd:abstract | //cit:abstract | dct:description |
| keywords |  | | |
| keywordsCodespace | | | |
| language | | | |
| externalId | | | |
| created | | | |
| updated | | | |
| publisher | | | |
| themes | | | |
| formats | | | |
| contactPoint | | | |
| license | | | |
| rights | | | |
| extent | | | |
| associations | | | |

Sources: https://github.com/SEMICeu/iso-19139-to-dcat-ap/blob/master/documentation/Mappings.md

**Comment by cholmes** [comment]
Awesome work. One thing that would be really great to add is what fields are required. That'll help me a lot in figuring out what to make required in the 'dataset record' proposal, see https://github.com/opengeospatial/ogcapi-records/pull/130

Also do any of these have default values? Like for language or keywordsCodespace?

**Comment by ByronCinNZ** [comment]
Working on that and will include indication of manditory ISO fields in my pull request

**Comment by ByronCinNZ** [comment]
Another Candidate new mapping from ISO19115-3
Metadata Contact //mdb:MD_Metadata  /mdb:contact  
Candidate geoJSON mapping - recordContact

**Comment by pvretano** [comment]
@ByronCinNZ what is the description ISO1911503 //mdb:MD_Metadata  /mdb:contact ?  Is it a contact responsible for the record itself ... as opposed to the resource that the record describes?

+1 @cholmes https://github.com/opengeospatial/ogcapi-code-sprint-2021-07/pull/23#issuecomment-885609782

**Comment by ByronCinNZ** [comment]
@pvretano Yes that is a contact for the record not the resource (which is captured separately)
